### DISCUSSION TITLE
심사 요청 진행 중 INVALID_SELLER_TOKEN, message=JWe decrypt exception, 오류가 발생합니다

### DISCUSSION THREAD

**Discussion by e34652** [commerce-api-naver/commerce-solutions#101] [묻고 답하기]
심사 요청을 위해 사용 시작 승인(/subscriptions/approve) 호출하기 전
판매자 인증 JWE 해석(/seller-info-by-token) 과정에 문제가 발생합니다

이전에 심사 요청을 눌러 스토어 ID로 로그인을 한 후 
리다이렉트된 URL에 있는 jwe token 값을 바탕으로 url을 생성 후 요청을 보냈습니다 

Bearer 토큰은 SELF 타입으로 발급 받았고
아래의 코드를 따라 진행했고 에러가 발생했습니다

        Request request = new Request.Builder()
                .url("https://api.commerce.naver.com/external/v1/commerce-solutions/seller-info-by-token?token=" + JWE 토큰값
                .get()
                .addHeader("Authorization", "Bearer SELF토큰값")
                .build();
                
        try {
            sellerAuthInfo = client.newCall(request).execute();
        } catch (IOException e) {
            throw new RuntimeException(e);
        }

code= 401 INVALID_SELLER_TOKEN, message=JWe decrypt exception
에러 메시지는 다음과 같았습니다

BEARER 토큰의 타입은 SELF(솔루션) 이었는데
왜 INVALID_SELLER_TOKEN 이라는 응답이 왔고

seller-info-by-token에 jwe 토큰값을 decrypt 하는 요청을 보냈는데 
왜 JWe decrypt exception이 발생했을까요?

제 이해나 방식이 잘못을까요? ㅠ
도움 주시면 감사하겠습니다

traceId = pr1-185309-69bxbz^1733910841346^176414241

**Comment by commerce-api-naver** [comment]
안녕하세요. 커머스API 이민석입니다.

JWE 토큰은 솔루션별로 다른 값이 생성됩니다. 또한 생성한 개발자 계정이나 솔루션 이름이 같더라도 솔루션 ID가 다른 경우 내부적으로 별개의 솔루션으로 취급됩니다.
정황상 운용중이신 다른 솔루션(A)으로 생성된 JWE 토큰 값을 또다른 솔루션(B)의 솔루션 애플리케이션으로 생성한 인증 토큰을 이용하여 복호화를 시도하신 것으로 보입니다.

JWE 토큰 값이 생성된 솔루션에 생성된 솔루션 애플리케이션으로 인증 토큰을 생성하여 다시 시도해보실 수 있을까요?
(참고로 대상 API의 인증 토큰은 알고계신 것과 같이 SELF 유형의 인증 토큰을 사용하심이 맞습니다.)

**Reply by e34652** [reply]
기존 솔루션과 동일한 솔루션을 자체 결제 방식으로 새로 등록하기 위해 심사 요청을 진행하고 있는데
현재 심사중인 솔루션을 위해 발급된 JWE 토큰을 기존 솔루션의 인증 토큰으로 복호화를 시도해서 문제가 되었다는 말씀이실까요? 
그럼 혹시 기존의 B 솔루션을 그대로 A 솔루션으로 활용하려면 어떻게 해야할까요? 
현재 토큰 발행이 ClientId와 Secret 기반으로 이루어지는데 이 값을 심사 요청 중인 솔루션의 것으로 교체하면 될까요?

**Reply by commerce-api-naver** [reply]
@e34652 님 안녕하세요. 커머스API 이민석입니다.

앞선 답변과 같이 동일한 개발사에서 만든 솔루션이더라도 각각의 솔루션의 정보는 철저히 분리하여 관리됩니다.
생성하신 솔루션은 상호간의 정보가 호환되지 않습니다. 완전히 별개의 솔루션으로 인지해주셔야 합니다.

A솔루션 구독 신청에 따라 생성된 JWE 토큰은 A솔루션의 솔루션 애플리케이션(A)으로만 복호화할 수 있습니다.
B솔루션 구독 신청에 따라 생성된 JWE 토큰은 B솔루션의 솔루션 애플리케이션(B)으로만 복호화할 수 있습니다.

2개 솔루션 간 '(심사 후 승인) Redirect URL'을 동일하게 지정하실 경우 어떤 솔루션에서 발생한 JWE 토큰인지 구분하기 어려울 수 있기에 URL 혹은 URL 파라미터등을 달리 지정하시어 JWE 토큰이 생성된 솔루션을 구분할 수 있도록 설계하신 후 각 JWE 토큰에 알맞은 솔루션 애플리케이션을 활용하도록 설계해보시기 바랍니다.

**Reply by e34652** [reply]
감사합니다 선생님 
리다이렉트 URL로 구분된다는 생각을 못하고 있었는데
말씀을 듣고 광명을 찾을 수 있었습니다

**Reply by commerce-api-naver** [reply]
@e34652 님 안녕하세요. 커머스API 이민석입니다.

대상 솔루션의 심사 요청과 관련하여 추가로 도움말을 드립니다.
아시다시피 보유하고 계신 솔루션 중 '자체 결제 방식(비즈월렛 미적용)' 솔루션만을 사용해주셔야 합니다.
심사 대상 솔루션(자체 결제 방식 적용 솔루션)이 아닌 다른 솔루션은 초기 연동사에 제공된 테스트용 솔루션이기에 가급적 사용하지 않는 것을 권해드리고 있습니다.
